src/coreclr/jit/importer.cpp

```diff
@@ -21688,15 +21536,12 @@ class SpillRetExprHelper
 
     void StoreRetExprResultsInArgs(GenTreeCall* call)
     {
-        for (GenTreeCall::Use& use : call->Args())
+        for (CallArg& arg : call->gtArgs.Args())
         {
-            comp->fgWalkTreePre(&use.NodeRef(), SpillRetExprVisitor, this);
+            comp->fgWalkTreePre(&arg.NodeRef(), SpillRetExprVisitor, this);
         }
 
-        if (call->gtCallThisArg != nullptr)
-        {
-            comp->fgWalkTreePre(&call->gtCallThisArg->NodeRef(), SpillRetExprVisitor, this);
-        }
+        // TODO-ARGS-REVIEW: Different order of this and rest of args
```

Comment:
@AndyAyersMS Was it intentional that the walk previously did not match evaluation order? Does it matter?

Reply:
IIRC @sandreenko added this -- I don't believe the order matters or the current order was intentional.

Reply:
Opened #67982